jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/TransferUtils.kt

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.s3
+
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.project.Project
+import com.intellij.util.io.size
+import software.amazon.awssdk.core.sync.RequestBody
+import software.amazon.awssdk.services.s3.S3Client
+import software.amazon.awssdk.services.s3.model.GetObjectRequest
+import software.amazon.awssdk.services.s3.model.GetObjectResponse
+import software.amazon.awssdk.services.s3.model.PutObjectRequest
+import software.amazon.awssdk.services.s3.model.PutObjectResponse
+import software.amazon.awssdk.utils.IoUtils
+import software.aws.toolkits.jetbrains.utils.ProgressMonitorInputStream
+import software.aws.toolkits.jetbrains.utils.ProgressMonitorOutputStream
+import software.aws.toolkits.resources.message
+import java.io.FileInputStream
+import java.io.FileOutputStream
+import java.io.InputStream
+import java.nio.file.Path
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionStage
+
+/**
+ * A set of utilities for uploading / downloading from S3 with progress indicators
+ */
+
+fun S3Client.upload(
+    project: Project,
+    source: Path,
+    bucket: String,
+    key: String,
+    message: String = message("s3.upload.object.progress", key),
+    startInBackground: Boolean = true
+): CompletionStage<PutObjectResponse> = upload(project, FileInputStream(source.toFile()), source.size(), bucket, key, message, startInBackground)
```

Comment:
We have `software.aws.toolkits.core.utils.PathUtilsKt#inputStream` so you dont need to do the toFile